### PR TITLE
fix: removed arrayCallbackFn

### PR DESCRIPTION
**Pull request by simlu**
BREAKING CHANGE:
Removed arrayCallbackFn. Was an edge case and hard to understand
function. Same functionality can be achieved using
breakFn and checking if the value / parent is an Array.

### CODE REVIEW

File: test/index.spec.js [hunk 1]
```diff
@@ -313,51 +313,6 @@ describe('Testing Find', () => {
...
-      });
-
-      it('Test arrayCallbackFn, not matched array', () => {
-        expect(tester(false, ['array.id'])).to.deep.equal({
-          default: ['array[0].id', 'array[1].id'],
-          array: []
-        });
-      });
-    });
-  });
```
Comment (simlu) [inline test/index.spec.js]: Restore and change to use breakfn?

Comment (simlu) [reply test/index.spec.js]: Added basic test now
